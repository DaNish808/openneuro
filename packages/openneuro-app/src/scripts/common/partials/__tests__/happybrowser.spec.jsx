import React from 'react'
import { shallow } from 'enzyme'
import Happybrowser from '../happybrowser'

const chrome =
  'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.78 Safari/537.36'
const chrome41 =
  'Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2272.104 Safari/537.36'
const chromium =
  'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/60.0.3112.113 Chrome/60.0.3112.113 Safari/537.36'
const googlebot =
  'Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)'
const ie =
  'Mozilla/5.0 (Windows; U; MSIE 6.0; Windows NT 5.1; SV1; .NET CLR 2.0.50727)'
const firefox =
  'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:57.0) Gecko/20100101 Firefox/57.0'

describe('common/partials/happybrowser', () => {
  it('renders successfully', () => {
    const wrapper = shallow(<Happybrowser ua={chrome} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders for unsupported browsers', () => {
    const wrapper = shallow(<Happybrowser ua={ie} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('is hidden in Chrome', () => {
    const wrapper = shallow(<Happybrowser ua={chrome} />)
    expect(wrapper.text()).toBe('')
  })
  it('is hidden in Chromium', () => {
    const wrapper = shallow(<Happybrowser ua={chromium} />)
    expect(wrapper.text()).toBe('')
  })
  it('is hidden when accessed by Googlebot', () => {
    const wrapper = shallow(<Happybrowser ua={googlebot} />)
    expect(wrapper.text()).toBe('')
  })
  it('is hidden when accessed by Chrome 41', () => {
    const wrapper = shallow(<Happybrowser ua={chrome41} />)
    expect(wrapper.text()).toBe('')
  })
  it('is visible in IE', () => {
    const wrapper = shallow(<Happybrowser ua={ie} />)
    expect(wrapper.text()).not.toBe('')
  })
  it('is hidden in Firefox', () => {
    const wrapper = shallow(<Happybrowser ua={firefox} />)
    expect(wrapper.text()).toBe('')
  })
})
