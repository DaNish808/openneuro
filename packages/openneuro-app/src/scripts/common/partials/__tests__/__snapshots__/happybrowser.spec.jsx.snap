// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/partials/happybrowser renders for unsupported browsers 1`] = `
<div
  className="happybrowser-markup"
>
  <div
    className="hb-wrap clearfix"
  >
    <div
      className="hb-text clearfix"
    >
      <img
        alt="warning"
        src="test-file-stub"
      />
      <p>
        We have detected that you are using an incompatible browser. This site may not work as expected.
         
        <strong>
          <a
            href="http://www.google.com/chrome/"
          >
            Please consider using a minimum of Chrome V49.0 or Firefox V50.0 as your browser
          </a>
          .
        </strong>
      </p>
    </div>
    <div
      className="hb-upgrade clearfix"
    >
      <div
        className="hb-img-wrap hb-dismiss"
        id="dismiss"
        onClick={[Function]}
      >
        X
      </div>
      <div
        className="hb-img-wrap hb-chrome"
      >
        <a
          href="http://www.google.com/chrome/"
        >
          <img
            alt="upgrade chrome"
            src="test-file-stub"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`common/partials/happybrowser renders successfully 1`] = `""`;
